osquery/tables/system/windows/bam.cpp

```diff
@@ -0,0 +1,97 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed in accordance with the terms specified in
+ *  the LICENSE file found in the root directory of this source tree.
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/system/time.h>
+#include <string>
+
+namespace osquery {
+namespace tables {
+
+constexpr auto kBamRegPath =
+    "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Services\\bam\\%%\\%%";
+
+// Get last exeution time
+auto last_execute_time(std::string& assist_data) {
+  std::string last_run_string = assist_data.substr(0, 16);
+
+  // Timestamp should always by 16 chars in length
+  if (last_run_string.length() == 16) {
+    // swap endianess
+    std::reverse(last_run_string.begin(), last_run_string.end());
+
+    for (std::size_t i = 0; i < last_run_string.length(); i += 2) {
+      char temp = last_run_string[i];
+      last_run_string[i] = last_run_string[i + 1];
+      last_run_string[i + 1] = temp;
+    }
+
+    // Convert Windows FILETIME to UNIX Time
+    unsigned long long last_run =
+        tryTo<unsigned long long>(last_run_string, 16).takeOr(0ull);
+    if (last_run == 0ull) {
+      LOG(WARNING) << "Failed to convert FILETIME to UNIX time.";
+      return std::string();
+    }
+    last_run = (last_run / 10000000) - 11644473600;
+    std::time_t last_run_time = last_run;
```

Comment:
Same as #5539, let's use the [`FILETIME` to a unix timestamp function](https://github.com/osquery/osquery/blob/a60b940290faf1e94dc66f5549f4d3568a0f9500/osquery/filesystem/windows/fileops.cpp#L1563) here. You'll also need to import <osquery/filesystem/fileops.h>

Reply:
Converted to FILTIME function in fileops.h